packages/expo-cli/src/commands/push-creds.js

```diff
@@ -72,6 +72,128 @@ export default (program: any) => {
 
       await apiClient.deleteAsync(`credentials/push/android/${remotePackageName}`);
 
+      log('All done!');
+    }, true);
+
+  const vapidSubjectDescription =
+    'URL or `mailto:` URL which provides a point of contact in case the push service needs to contact the message sender.';
+
+  program
+    .command('push:web:upload [project-dir]')
+    .description('Uploads VAPID key pair and VAPID subject for web push notifications.')
+    .option('--vapid-pubkey [vapid-public-key]', 'URL-safe base64-encoded VAPID public key.')
+    .option('--vapid-pvtkey [vapid-private-key]', 'URL-safe base64-encoded VAPID private key.')
+    .option('--vapid-subject [vapid-subject]', vapidSubjectDescription)
+    .asyncActionProjectDir(async (projectDir, options) => {
+      if (!options.vapidPubkey || !options.vapidPvtkey || !options.vapidSubject) {
+        throw new Error(
+          'Must specify all three fields (--vapid-pubkey, --vapid-pvtkey, and --vapid-subject) to upload.'
+        );
+      }
+
+      log('Reading project configuration...');
+
+      const {
+        args: { remotePackageName },
+      } = await Exp.getPublishInfoAsync(projectDir);
+
+      log('Logging in...');
+
+      const user = await UserManager.getCurrentUserAsync();
+      const apiClient = ApiV2.clientForUser(user);
+
+      log("Setting VAPID keys on Expo's servers...");
+
+      await apiClient.putAsync(`credentials/push/web/${remotePackageName}`, {
+        vapidPublicKey: options.vapidPubkey,
+        vapidPrivateKey: options.vapidPvtkey,
+        vapidSubject: options.vapidSubject,
+      });
+
+      log('All done!');
+    }, true);
+
+  program
+    .command('push:web:generate [project-dir]')
+    .description('Generates VAPID key pair for web push notifications.')
+    .option('--vapid-subject [vapid-subject]', vapidSubjectDescription)
+    .asyncActionProjectDir(async (projectDir, options) => {
+      if (!options.vapidSubject) {
+        throw new Error('Must specify --vapid-subject.');
+      }
+
+      log('Reading project configuration...');
+
+      const {
+        args: { remotePackageName },
+      } = await Exp.getPublishInfoAsync(projectDir);
+
+      log('Logging in...');
+
+      const user = await UserManager.getCurrentUserAsync();
+      const apiClient = ApiV2.clientForUser(user);
+
+      log("Generating and setting VAPID keys on Expo's servers...");
+
+      const results = await apiClient.putAsync(`credentials/push/web/${remotePackageName}`, {
+        vapidSubject: options.vapidSubject,
+      });
+
+      log('All done!');
+      log(`Your VAPID public key is: ${results.vapidPublicKey}`);
+      log(`Your VAPID private key is: ${results.vapidPrivateKey}`);
+      log(
+        "Don't forget to also add the generated public key to your `app.json` file! Learn more here: https://docs.expo.io/versions/latest/guides/using-vapid/"
```

Comment:
one thought -- it might be nice to just automatically modify the app.json file to have this key. we do this in a few other places, e.g. in `eject` we add the bundleIdentifier/packageName after prompting for it, and it's always nice to save the developer little busywork steps where we can.

Reply:
I've added this as well as warnings. :)